daemon/access.go

```diff
@@ -246,3 +249,48 @@ func (ac interfaceAuthenticatedAccess) CheckAccess(d *Daemon, r *http.Request, u
 
 	return Unauthorized("access denied")
 }
+
+// noticesReadAccess allows mapping notice types to access checkers and otherwise
+// behaves like openAccess (ie. allows access from snapd.socket only).
+type noticesReadAccess struct {
+	Checkers       map[state.NoticeType]accessChecker
+	FindNoticeByID bool
+}
+
+func (ac noticesReadAccess) CheckAccess(d *Daemon, r *http.Request, ucred *ucrednet, user *auth.UserState) *apiError {
+	var types []state.NoticeType
+	var err error
+
+	if ac.FindNoticeByID {
+		// GET /v2/notices/{id}
+		noticeID := muxVars(r)["id"]
+		st := d.overlord.State()
+		st.Lock()
+		notice := st.Notice(noticeID)
```

Comment:
maybe we should do what @olivercalder suggests but not quite:

* the checkers should not look at type, but just allow access to /v2/notices /v2/notices/{id} for a set of interfaces
* inside the code /v2/notices we filter out not allowed types by interface, in /v2/notices/{id} we fail if the type is not allowed

the annoying bit of this is finding about the interface twice, but in principle we could attach the interface to r.RemoteAddr and have a variant of ucrednetGet that extracts that as well as the *ucrednet